locust/runners.py

```diff
@@ -49,6 +51,27 @@ def on_hatch_complete(user_count):
                 self.stats.reset_all()
         events.hatch_complete += on_hatch_complete
 
+    @staticmethod
+    def build_locust_classes_by_name(locust_classes):
+        """
+        Returns a dictionary of the form {locust_class_name: locust_class}
+        """
+        if isinstance(locust_classes, type):
+            return {locust_classes.__name__: locust_classes}
+        elif isinstance(locust_classes, collections.Iterable):
+            return {locust.__name__: locust for locust in locust_classes}
+        else:
+            return {}
+
+    def filter_locusts_by_name(self, locust_class=None):
+        """
+        By default, get all locusts. If locust_class is supplied, return a list of
+        only a locust with that class name. If it doesn't exist, return all locusts
+        """
+        if locust_class and locust_class in self.locust_classes_by_name:
```

Comment:
would it be better to throw an exception if someone passes an invalid locust_class name rather than silently returning all?

Reply:
Yeah, I'll do that.